web_widget_x2many_2d_matrix/static/src/js/widget_x2many_2d_matrix.js

```diff
@@ -63,6 +63,8 @@ odoo.define('web_widget_x2many_2d_matrix.widget', function (require) {
                 this.parse_boolean(node.show_row_totals || '1');
             this.show_column_totals =
                 this.parse_boolean(node.show_column_totals || '1');
+            this.label_cell_max_char = this.nodeOptions.label_cell_max_char || -1;
+            this.header_max_char = this.nodeOptions.header_max_char || -1;
```

Comment:
If I remember correctly, the CSS would not apply and the column would get quite large or split on many lines.

It will not get fixed so closing this.